src/Data/Enum.purs

```diff
@@ -242,3 +242,10 @@ defaultToEnum n
   -- | Runs in `O(n)` where `n` is `fromEnum a`
 defaultFromEnum :: forall a. Enum a => a -> Int
 defaultFromEnum = maybe 0 (\prd -> defaultFromEnum prd + 1) <<< pred
+
+-- | Like `toEnum` but returns `bottom` if `x` is less than `fromEnum bottom`
+-- | and `top` if `x` is greater than `fromEnum top`.
+toBoundedEnum :: forall a. (BoundedEnum a) => Int -> a
+toBoundedEnum x = case toEnum x of
+  Just enum -> enum
+  Nothing -> if x < fromEnum (bottom :: a) then bottom else top
```

Comment:
I went with this approach because it only requires a single comparison. If I do the comparison before `toEnum`, I have to check both `x < bottom` and `x > top`. 

Alternatively, I could use `clamp`.